pilot/pkg/model/config_test.go

```diff
@@ -177,42 +178,70 @@ func TestServiceKey(t *testing.T) {
 func TestSubsetKey(t *testing.T) {
 	hostname := model.Hostname("hostname")
 	cases := []struct {
+		got      string
 		hostname model.Hostname
 		subset   string
 		port     *model.Port
 		want     string
+		err      error
 	}{
 		{
 			hostname: "hostname",
 			subset:   "subset",
 			port:     &model.Port{Name: "http", Port: 80, Protocol: model.ProtocolHTTP},
-			want:     "outbound|http|subset|hostname",
+			want:     "outbound.http.subset.hostname",
 		},
 		{
 			hostname: "hostname",
 			subset:   "subset",
 			port:     &model.Port{Port: 80, Protocol: model.ProtocolHTTP},
-			want:     "outbound||subset|hostname",
+			want:     "outbound.80.subset.hostname",
```

Comment:
This is wrong. Eds will fail if you use port numbers in cluster names because the k8s registry code is hardwired to use port names to query everything.

Reply:
Good point - I think we can fix EDS ? We could also leave it empty ("_").

Reply:
This uses port number if port name is not specified.
What is the general plan if port name is not specified ?